akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -317,6 +317,26 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Limit[T](n: Int) extends PushPullStage[T, T] {
+  private var left = n
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
+    left -= 1
+    if (left >= 0) ctx.push(elem)
+    else {
+      if (ctx.isFinishing) ctx.push(elem)
+      else ctx.fail(new Exception("more elements not allowed"))
```

Comment:
this needs to be probably an OutOfBoundException. Also, it should contain a more informative text like "More than $n elements arrived."